enhancements/authentication/separate-oauth-resources.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Separate-OAuth-API-Resources
+authors:
+  - "@deads2k"
+reviewers:
+  - "@sttts"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-10-16
+last-updated: 2019-10-16
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Separate OAuth API-Resources
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+The API resources used by the OpenShift OAuth server will be moved to their own `openshift-oauth-apiserver` repository.
+The openshift-oauth-apiserver binary will be managed by the existing authentication-operator.
+
+## Motivation
+
+The OpenShift OAuth server requires several openshift API resources in order to function.
+Because the OAuth server enables access to the kube-apiserver, it needs to be able to run separately from the existing
+set of API resources that are focused on enabling a developer workflow.
+This narrowing of scope expands the deployment options for the OpenShift OAuth server.
+
+### Goals
+
+1. Make it possible to run the OpenShift OAuth server without installing all the developer focused API Resources.
+
+### Non-Goals
+
+1. Move routes out of the OpenShift API Server.
+
+## Proposal
+
+1. Create a new repo called `openshift-oauth-apiserver` that produces a binary called `openshift-oauth-apiserver`.
+2. Add the following API groups to the new repo
```

Comment:
Put another way, if you had a different oauth server, would you want to keep authorization, project, quota, and security?  Maybe.

Reply:
I'm not sure to which degree this is a rhetorical question, but to make sure I will just say that I am missing the direct connection between oauth and the authorization groups below (namely SCCs, projects and quotas, I understand that we would like to have the authorization group kept somewhere and leave it to die).

It's fine to have the authentication operator take care of these resources (via apiserver + apiservice), and the operator should also be the config endpoint that would allow you use of an external oauth server, in which case I'd say you would still be able to keep access to these resources even if you use your own oauth server.

If these resources should live in one apiserver, I'd probably rather call it `openshift-auth-apiserver` given the fairly vague connection between oauth and the groups I mentioned earlier.

Reply:
The question is no longer relevant given the most recent changes